website/sales/api/v2/views.py

```diff
@@ -111,4 +108,35 @@ def destroy(self, request, *args, **kwargs):
             raise PermissionDenied
         if self.get_object().payment:
             raise PermissionDenied
-        return super(UserOrderDetailView, self).destroy(request, *args, **kwargs)
+
+
+class OrderClaimView(RetrieveAPIView):
+    """Claims an order to be paid by the current user."""
+
+    queryset = Order.objects.all()
+    serializer_class = UserOrderSerializer
+    schema = AutoSchema(operation_id_base="claimOrder")
+    permission_classes = [IsAuthenticatedOrTokenHasScope]
+    required_scopes = ["sales:order"]
+
+    def retrieve(self, request, *args, **kwargs):
```

Comment:
I don't think we should do this on a `GET`, as I wouldn't say it good practice to do any writing on a `GET` (especially not something that is irreversible).